test/statefulset-test.js

```diff
@@ -0,0 +1,179 @@
+'use strict';
+
+const test = require('tape');
+const nock = require('nock');
+
+const openshiftRestClient = require('../');
+const privates = require('../lib/private-map');
+
+const settings = {
+  config: {
+    apiVersion: 'v1beta1',
+    context:
+     { cluster: '192-168-99-100:8443',
+       namespace: 'for-node-client-testing',
+       user: 'developer/192-168-99-100:8443' },
+    user: { token: 'zVBd1ZFeJqEAILJgimm4-gZJauaw3PW4EVqV_peEZ3U' },
+    cluster: 'https://192.168.99.100:8443' }
+};
+
+test('find - statefulsets - basic findAll', (t) => {
+  openshiftRestClient(settings).then((client) => {
+    t.equal(typeof client.statefulsets.findAll, 'function', 'There is a findAll method on the statefulsets object');
+
+    const clientConfig = privates.get(client).config;
+
+    nock(clientConfig.cluster)
+      .matchHeader('authorization', `Bearer ${clientConfig.user.token}`) // taken from the config
+      .get(`/apis/apps/v1beta1/namespaces/${clientConfig.context.namespace}/statefulsets`)
+      .reply(200, {kind: 'StatefulSetList'});
+
+    const findResult = client.statefulsets.findAll().then((statefulSetList) => {
+      t.equal(statefulSetList.kind, 'StatefulSetList', 'returns an object with StatefulSetList');
+      t.end();
+    });
+
+    t.equal(findResult instanceof Promise, true, 'should return a Promise');
+  });
+});
+
+test('find - statefulsets - basic find', (t) => {
+  openshiftRestClient(settings).then((client) => {
+    t.equal(typeof client.statefulsets.find, 'function', 'There is a find method on the statefulsets object');
+
+    const clientConfig = privates.get(client).config;
+    const statefulSetName = 'cool-statefulset-name-1';
+
+    nock(clientConfig.cluster)
+      .matchHeader('authorization', `Bearer ${clientConfig.user.token}`) // taken from the config
+      .get(`/oapi/v1/namespaces/${clientConfig.context.namespace}/statefulsets/${statefulSetName}`)
+      .reply(200, {kind: 'StatefulSet'});
+
+    const findResult = client.statefulsets.find(statefulSetName).then((statefulset) => {
+      t.equal(statefulset.kind, 'StatefulSet', 'returns an object with StatefulSet');
+      t.end();
+    }).catch(e => {
+      console.log(e);
+    });
+
+    t.equal(findResult instanceof Promise, true, 'should return a Promise');
+  });
+});
+
+test('find - statefulsets - find - no statefulset name', (t) => {
+  openshiftRestClient(settings).then((client) => {
+    client.statefulsets.find().catch((err) => {
+      t.equal(err.message, 'StatefulSet Name is required', 'error message should return');
```

Comment:
Tests fail because it is looking for `Stateful Set`

Reply:
Again...stupid mistake:) Thanks a lot for review. I am about to push an update.